agents/components/Profile.js

```diff
@@ -1,118 +1,109 @@
-import { pipe } from 'ramda'
+import { pipe, isNil, not } from 'ramda'
 import { TextField } from 'redux-form-material-ui'
 import RaisedButton from 'material-ui/RaisedButton'
+import { compose, withState, withHandlers } from 'recompose'
+import h from 'react-hyperscript'
 
 import { FormattedMessage } from '../../lib/Intl'
 import styles from '../styles/Profile'
 import Button from '../../app/components/Button'
 import AvatarField from '../../app/components/AvatarField'
 
-class Profile extends React.Component {
-  constructor (props, context) {
-    super(props, context)
-    this.state = {
-      isEditing: false
-    }
-  }
-
-  toggleEdit () {
-    this.setState({
-      isEditing: !this.state.isEditing
-    })
-  }
-
-  render () {
-    const { isEditing } = this.state
-    const { styles, agent, agent: { profile: { name, description, avatar } } } = this.props
-
-    return (
-      <form className={styles.container}>
-        <p className={styles.intro}>
-          <FormattedMessage
-            id='agents.profile'
-            className={styles.labelText}
-          />
-        </p>
-        <div className={styles.innerContainer}>
-          <div className={styles.avatarContainer}>
-            <Field
-              name='avatar'
-              component={AvatarField}
-              isEditingProfile={isEditing}
-              value={avatar}
-            />
-          </div>
-          <div className={styles.infoContainer}>
-            <Field
-              name='name'
-              floatingLabelText={
-                <FormattedMessage
-                  id='agents.nameLabel'
-                  className={styles.labelText}
-                />
-              }
-              component={TextField}
-              fullWidth={true}
-              value={name}
-              disabled={!isEditing}
-            />
-            <Field
-              name='description'
-              floatingLabelText={
-                <FormattedMessage
-                  id='agents.descriptionLabel'
-                  className={styles.labelText}
-                />
-              }
-              component={TextField}
-              value={description}
-              fullWidth={true}
-              multiLine={true}
-              rowsMax={5}
-              disabled={!isEditing}
-            />
-          </div>
-        </div>
-        <div className={styles.buttonContainer}>
-          <RaisedButton className={styles.button} type='button' onClick={() => { this.toggleEdit() }}>
-            {
-              isEditing
-              ? <FormattedMessage
-                  id='agents.saveProfile'
-                  className={styles.labelText}
-                />
-              : <FormattedMessage
-                  id='agents.editProfile'
-                  className={styles.labelText}
-                />
-            }
-          </RaisedButton>
-        </div>
-      </form>
-    )
-  }
-
-}
-
-Profile.propTypes = {
```

Comment:
i think we've decided to not use them. but also still keen on a type api, just haven't made it a priority in our sprints to add type schemas, which maybe we should do.